purchase_vendor_promotion/models/stock_route.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockRoute(models.Model):
+    _inherit = "stock.route"
+
+    is_force_default_vendor = fields.Boolean(default=False)
```

Comment:
I would suggest more explicit name: `force_vendor_with_best_promotion`

Would also need a `data/stock_route.xml` file to update our Buy route:

```
        <record id="purchase_stock.route_warehouse0_buy" model="stock.route">
            <field name="force_vendor_with_best_promotion">True</field>
        </record>
   ```

Reply:
I guess you suggest to add the data file to allo's local module right (like `local-src/allo_data_stock`) ?

Reply:
hi, I think it makes sense to add here directly as it goes well with the global purpose of the module

Reply:
I agree, with explicit description maybe to let users aware of this feature